packages/devtools_app/lib/src/tree.dart

```diff
@@ -56,21 +57,30 @@ class _TreeViewState<T extends TreeNode<T>> extends State<TreeView<T>>
   @override
   void initState() {
     super.initState();
-    _initData();
-    _updateItems();
+    widget.dataRootsListenable.addListener(_updateTreeView);
+    _updateTreeView();
   }
 
   @override
-  void didUpdateWidget(TreeView<T> oldWidget) {
-    super.didUpdateWidget(oldWidget);
-    if (widget.dataRoots != oldWidget.dataRoots) {
-      _initData();
-      _updateItems();
-    }
+  void dispose() {
+    widget.dataRootsListenable.removeListener(_updateTreeView);
+    super.dispose();
   }
 
+  // @override
+  // void didUpdateWidget(TreeView<T> oldWidget) {
+  //   super.didUpdateWidget(oldWidget);
+  //   _initData();
+  //   _updateItems();
+  // }
+
   void _initData() {
-    dataRoots = List.from(widget.dataRoots);
+    dataRoots = List.from(widget.dataRootsListenable.value);
```

Comment:
just put this directly in _updateTreeView since _initData is not used anywhere else

Reply:
Done